restli-tools/src/main/resources/macros/library.vm

```diff
@@ -169,21 +177,34 @@
   #end
 #end
 
+#macro(assocKeyParams $method $includeOptional)
+    #set($hasMoreParams = $method.hasRequiredParams() || ($includeOptional && ($method.hasOptionalParams() || $method.hasProjectionParams())))
+    #foreach($assocKey in $method.assocKeys)
+      $assocKey.declaredType $assocKey.name#if( $foreach.hasNext || $hasMoreParams),#end
```

Comment:
I actually think it should be "assocKey.bindingType" here. Can you please check?